Readme.md

```diff
@@ -18,18 +18,46 @@ This is an ipm package, which contains one or more reusable assets within the ip
 [Browse ipm Packages](https://ipm.clearblade.com)
 
 ## Setup
-If you already have a GCloud account and the CLI set up, API key can be generated/accessed by going to this page: https://cloud.google.com/bigquery/docs/authorization
- 
-Another simple way: Copy and paste the following command 'gcloud auth application-default print-access-token' in Google Cloud Platform shell!
 
+### 1. API Key
 
-## API<a name="BigQuery"></a>
+Access an API Key here:
+https://cloud.google.com/bigquery/docs/authorization
 
-## Typedefs
+If you have the GCloud CLI, can obtain it by running
+
+```
+gcloud auth application-default print-access-token
+```
+
+### 2. Dataset
+
+- Create a new project
+- Create a new dataset in that project by loading custom dataset following the procedure given [here](https://cloud.google.com/bigquery/quickstart-web-ui#download_custom_data). 
+
+## Assets
+
+## Code Libraries
+
+- `BigQueryLib` - Library for interacting with GCloud BigQuery Dataset
+
+### Code Services
+
+ - `ExampleDeleteDataset` - delete a dataset.
```

Comment:
I know, we might have decided later that services names should be **PackageShortName**ExampleServiceName